packages/web-components/src/components/cbp-toggle/cbp-toggle.tsx

```diff
@@ -39,18 +39,46 @@ export class CbpToggle {
     });
   }
 
+  @Listen('keydown')
+  handleKeyDown( ev: KeyboardEvent){
+    if(ev.key === ' '){
+      ev.preventDefault(); //techdebt: needed to prevent scrolling when using this keypress
+      this.toggleEvent();
+    }
+  }
+
+  @Event() toggleClick: EventEmitter;
+
   /** Event: toggles the control true/false & updates DOM accordingly*/
   toggleEvent(){
-    //TODO: logic here
+    let checkbox = this.host.querySelector('input[type="checkbox"]');
+
+    if(this.checked && !this.disabled){
+      this.checked = false;
+      checkbox.setAttribute('checked', 'false');
+    }else if(!this.disabled) {
+      this.checked = true;
+      checkbox.setAttribute('checked', 'true');
+    }
   }
 
   render() {
-    /**boilerplate HTML */
-    return (
-      <Host>
-        <slot></slot>
-      </Host>
-    );
+    let ariaLabel = this.host.querySelector('label').textContent;
+    let ariaCheck = this.checked;
+
+      return (
+        <Host
```

Comment:
Now that you're using a real checkbox as the control, this should not have role=checkbox or tabindex (I notice 2 tab stops and 1 had not visible focus indicator). You shouldn't need the click handler because everything should be rendered in a label like a checkbox. See my latest checkbox updates for generating and applying the fieldId (also allows it to be specified on the slotted control instead). Adding a label means you'll need to move the flex context to it instead. You also shouldn't need the named slot for the label. (I've done some testing and flex seems to work on plain text nodes as if it was an element - if it gives you trouble, let's screen share and discuss).

Reply:
Updated as discussed